examples/js/nodes/NodeMaterialSource.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * @author sunag / http://www.sunag.com.br/
+ */
+
+( function () {
+	
+	function loadScript( path ) {
+
+		var js = document.createElement("script");
+		js.type = 'text/javascript';
+		js.src = path;
+
+		document.body.appendChild( js );
+
+	}
+	
+	function loadScripts( path, scripts ) {
+
+		for(var i = 0; i < scripts.length; i++) {
+			
+			loadScript( path + scripts[i] );
+			
+		}
+
+	}
+	
+	loadScripts( "js/nodes/", [
+		// NodeLibrary
```

Comment:
I think we may want to turn all the nodes into ES6 Modules and let the browser resolve the dependencies for us 🙂

Reply:
> I think we may want to turn all the nodes into ES6 Modules and let the browser resolve the dependencies for us 

Loading large numbers of modules in the browser will be slower than bundling

![renderer-main-thread-time-breakdown](https://user-images.githubusercontent.com/5307958/41634331-74c02a98-746d-11e8-9303-a9449f952f6b.png)


From this article: https://developers.google.com/web/fundamentals/primers/modules#performance

Reply:
@mrdoob What do you think of including `build/three.nodes.js` as extesion?

Reply:
> Loading large numbers of modules in the browser will be slower than bundling

I think it's too early to think about bundling. Besides, currently we're including scripts separately and adding dependency logic to the code.

I'm suggesting porting it to ES6 Modules so it'll be easier to integrate in src/ later on.

Anyway, this can be done in a different PR.

Reply:
> Yes! I had already converted to ES6 but the code turned out to be outdated. You have some example that I can base?

https://github.com/mrdoob/three.js/blob/dev/examples/webgl2_sandbox.html uses ES6 Modules in the example directly 😊

Reply:
> https://github.com/mrdoob/three.js/blob/dev/examples/webgl2_sandbox.html uses ES6 Modules in the example directly blush

Niice!